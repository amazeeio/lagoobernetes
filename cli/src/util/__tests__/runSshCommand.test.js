// @flow

import execa from 'execa';
import { runSshCommand } from '../runSshCommand';

jest.mock('execa');
jest.mock('../../config/getSshConfig', () => ({
  getSshConfig: () => ({ username: 'lagoobernetes', host: 'localhost', port: 2020 }),
}));

// Flow does not know which objects are actual mocks
// this function casts given parameter to JestMockFn
const _castMockForFlow = (mockFn: any): JestMockFn<any, any> => mockFn;

const mockedExeca = _castMockForFlow(execa);
const stdout = 'mocked stdout content\nsecondline';

mockedExeca.mockImplementation(async () => ({
  stdout,
}));

describe('runSshCommand', () => {
  it('should call execa with correct arguments and return stdout', async () => {
    const returned = await runSshCommand({
      command: 'token',
      identity: './identity/path',
    });
    const execaCalls = mockedExeca.mock.calls;
    expect(execaCalls.length).toBe(1);
    expect(execaCalls).toMatchSnapshot();
    expect(returned).toBe(stdout);
  });

  it('should throw stderr on error', async () => {
    const stderrContent = 'mocked stderr content\nanother second line';
    mockedExeca.mockImplementationOnce(async () => {
      throw {
        stderr: stderrContent,
      };
    });
    try {
      await runSshCommand({
        command: 'token',
        identity: './identity/path',
      });
    } catch (err) {
      expect(err).toBe(stderrContent);
    }
  });
});
