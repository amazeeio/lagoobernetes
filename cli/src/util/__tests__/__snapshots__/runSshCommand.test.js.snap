// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runSshCommand should call execa with correct arguments and return stdout 1`] = `
Array [
  Array [
    "ssh",
    Array [
      "-p",
      "2020",
      "-i",
      "./identity/path",
      "-F",
      "/dev/null",
      "-o",
      "AddKeysToAgent=no",
      "-o",
      "IdentitiesOnly=yes",
      "lagoobernetes@localhost",
      "token",
    ],
  ],
]
`;
